components/feature/media/src/main/java/mozilla/components/feature/media/service/MediaSessionServiceDelegate.kt

```diff
@@ -116,7 +125,13 @@ internal class MediaSessionServiceDelegate(
         }
 
         updateMediaSession(state)
-        updateNotification(state)
+        MainScope().launch() {
```

Comment:
You are still creating a scope per launch that never gets cancelled.

You may want to use something like this:

```Kotlin
class ... {
    val scope: CoroutineScope? = null
    
    fun onCreate() {
        scope = MainScope()
    }

    fun onDestroy() {
        scope?.cancel()
    }
}
```

This way all coroutines launched here are bound to the scope/lifecycle of the service. If the service gets destroyed all scheduled coroutines get stopped too.

Reply:
Understood.  I'll update